src/_guides/language/coming-from/js-to-dart.md

```diff
@@ -2356,27 +2519,68 @@ consumer = Cat();
 consumer.consume(); // Eating mice...
 ```
 
-### Mixins 
+### Mixins
 
 Mixins are used to share functionality between classes.
 You can use the mixin's fields and methods in the class,
 using their functionality as if it were part of the class.
-A class can use multiple mixins, which is useful
-when multiple classes share the same functionality,
+A class can use multiple mixins. This helps when multiple classes share the
+same functionality,
 without needing to inherit from each other or share a common ancestor.
 
-A mixin is declared like a regular class,
-as long as it doesn't extend any class other
-than `Object` and has no constructors.
-Use the `with` keyword to add one or more
-comma-separated mixins to a class.
-Although JavaScript doesn’t have an equivalent
-for this keyword, the effect is similar to using
-`Object.assign` to merge additional objects into
-an existing object, after instantiating.
+Use the `with` keyword to add one or more comma-separated mixins to a class.
+
+JavaScript has no keyword equivalent. JavaScript can use `Object.assign`
+to merge additional objects into an existing object, after instantiating.
+
+The following examples show how JavaScript and Dart achieve similar behavior:
+
+
+{:.include-lang}
+```js
+class Animal {}
+
+// Defining the mixins
+class Flyer {
+  fly = () => console.log('Flaps wings');
+}
+class Walker {
+  walk = () => console.log('Walks legs');
+}
+ 
+class Bat extends Animal {}
+class Goose extends Animal {}
+class Dog extends Animal {}
+
+// Composing the class instances with
+// their correct functionality.
+const bat =
+  Object.assign(
+    new Bat(),
+    new Flyer()
```

Comment:
This works, but it relies on the `fly`/`walk` "methods" being declared as fields instead.
If you wrote it as 
```js
class Flyer {
  fly() => console.log('Flaps wings');
}
```
instead, which would be the canonical way to write a JS class, the `assign` would not copy the method (because it's on the prototype and is not enumerable).
Might be a caveat worth mentioning.

Reply:
As this describes a specific caveat for JS, I'll leave it as is.